Documentation/cmdref/cilium-agent.md

```diff
@@ -75,11 +75,13 @@ cilium-agent [flags]
       --ipsec-key-file string                      Path to IPSec key file
       --ipv4-cluster-cidr-mask-size int            Mask size for the cluster wide CIDR (default 8)
       --ipv4-node string                           IPv4 address of node (default "auto")
+      --ipv4-pod-subnets strings                   List of IPv4 pod subnets to preconfigure for encryption
       --ipv4-range string                          Per-node IPv4 endpoint prefix, e.g. 10.16.0.0/16 (default "auto")
       --ipv4-service-loopback-address string       IPv4 address for service loopback SNAT (default "169.254.42.1")
       --ipv4-service-range string                  Kubernetes IPv4 services CIDR if not inside cluster prefix (default "auto")
       --ipv6-cluster-alloc-cidr string             IPv6 /64 CIDR used to allocate per node endpoint /96 CIDR (default "f00d::/64")
       --ipv6-node string                           IPv6 address of node (default "auto")
+      --ipv6-pod-subnets strings                   List of IPv6 pod subnets to preconfigure for encryption
```

Comment:
To confirm, these options are global to the cluster on which Cilium is running? E.g. all cilium pods will be aware of all of the same subnets?

How does this work with say, node scale up / scale down events in a cluster? Will the subnets change?

Reply:
Correct these are global subnets where pod IPs are assigned from so any node may have an IP from one or all of these subnets.

In this mode rules are not added/removed on node events rather we preconfigure the network to work up-front. If the subnets change while scaling then the agent will need to be restarted with the new subnet ranges. If wanted a future feature could do this without a restart via CLI.

Reply:
OK, got it - thanks for confirming!